sr_data_visualization/scripts/sr_data_visualization/sr_data_visualizer_gui.py

```diff
@@ -0,0 +1,1095 @@
+
+        self.change_graphs(all=True, type="motor_stat", ncol=1)
+
+        self.t0 = 0
+        self.init_complete = True
+        # TODO: refresh graphs on resize?
+
+    def on_resize_main(self, empty):
+        if (self._widget.width() * self._widget.height()) < 3500000:
+            self.font_offset = -3
+        else:
+            self.font_offset = 1
+
+    def include_tactile_plugin(self):
+        self.tactile_gui_list = []
+        i = 0
+        self.timer = QTimer(self._widget)
+        while i < 5:
+            tactile_gui = SrGuiBiotac(None)
+            tactile_gui._widget = self._widget
+            tactile_gui.find_children(i)
+            self.tactile_gui_list.append(tactile_gui)
+            widget = self._widget.findChild(QWidget, "widget" + "_" + str(i))
+            widget.resizeEvent = self.on_resize_tactile
+
+            self.timer.timeout.connect(widget.update)
+            widget.paintEvent = tactile_gui.paintEvent
+            if not tactile_gui._hand_parameters.mapping:
+                rospy.logerr("No hand detected")
+            i += 1
+
+        self.label_list = []
+        self.lcd_list = []
+
+        j = 0
+        while j < 5:
+            i = 1
+            lcds = []
+            while i < 25:
+                if i < 10:
+                    lcd = self._widget.findChild(QLCDNumber, "lcdE0" + str(i) + "_" + str(j))  # %d_0" + "_" + str(i))
+                else:
+                    lcd = self._widget.findChild(QLCDNumber, "lcdE" + str(i) + "_" + str(j))  # %d_0" + "_" + str(i))
+                lcds.append(lcd)
+                i += 1
+            self.lcd_list.append(lcds)
+
+            i = 1
+            labels = []
+            while i < 25:
+                if i < 10:
+                    label = self._widget.findChild(QLabel, "label_E0" + str(i) + "_" + str(j))  # %d_0" + "_" + str(i))
+                else:
+                    label = self._widget.findChild(QLabel, "label_E" + str(i) + "_" + str(j))  # %d_0" + "_" + str(i))
+                labels.append(label)
+                i += 1
+            self.label_list.append(labels)
+            j += 1
+
+        self.timer.start(50)
+
+    def on_resize_tactile(self, none):
+        if (time.time() - self.t0) > 0:
+            for tactile_widget in self.tactile_gui_list:
+                tactile_widget.redraw_electrodes()
+
+            if (self._widget.width() * self._widget.height()) < 3500000:
+                scale = 6
+                mheight = 18
+            else:
+                scale = 12
+                mheight = 30
+
+            font = QFont('Sans Serif', scale)
+            font.setKerning(True)
+            j = 0
+            while j < 5:
+                i = 0
+                while i < 24:
+                    self.lcd_list[j][i].setMinimumHeight(2)
+                    self.lcd_list[j][i].setMaximumHeight(mheight)
+                    i += 1
+                i = 0
+                while i < 24:
```

Comment:
can't this loop be inside the other?

Reply:
Yup, well spotted, found another one too